src/app/clusters/diagnostic-logs-server/CodegenIntegration.cpp

```diff
@@ -0,0 +1,120 @@
+
+// Diagnostic logs cluster is a device-wide cluster, so we only need one cluster instance.
+LazyRegisteredServerCluster<DiagnosticLogsCluster> gDiagnosticLogsCluster;
+DiagnosticLogsProviderDelegate * gDiagnosticLogsProviderDelegate = nullptr;
+
+class IntegrationDelegate : public CodegenClusterIntegration::Delegate
+{
+public:
+    ServerClusterRegistration & CreateRegistration(EndpointId endpointId, unsigned emberEndpointIndex,
+                                                   uint32_t optionalAttributeBits, uint32_t featureMap) override
+    {
+        gDiagnosticLogsCluster.Create(endpointId);
+        if (gDiagnosticLogsProviderDelegate != nullptr)
+        {
+            gDiagnosticLogsCluster.Cluster().SetDelegate(gDiagnosticLogsProviderDelegate);
+        }
+        return gDiagnosticLogsCluster.Registration();
+    }
+
+    ServerClusterInterface & FindRegistration(unsigned emberEndpointIndex) override { return gDiagnosticLogsCluster.Cluster(); }
+    void ReleaseRegistration(unsigned emberEndpointIndex) override { gDiagnosticLogsCluster.Destroy(); }
+};
+
+} // namespace
+
+void emberAfDiagnosticLogsClusterServerInitCallback(EndpointId endpoint)
+{
+    IntegrationDelegate integrationDelegate;
+
+    CodegenClusterIntegration::RegisterServer(
+        {
+            .endpointId = endpoint,
+            .clusterId  = DiagnosticLogs::Id,
+            .fixedClusterServerEndpointCount =
+                1, // Diagnostic logs are device-wide functionality, so we only need one cluster instance
+            .maxEndpointCount        = 1,
+            .fetchFeatureMap         = false,
+            .fetchOptionalAttributes = false,
+        },
+        integrationDelegate);
+}
+
+void MatterDiagnosticLogsClusterServerShutdownCallback(EndpointId endpointId)
+{
+    IntegrationDelegate integrationDelegate;
+
+    CodegenClusterIntegration::UnregisterServer(
+        {
+            .endpointId = endpointId,
+            .clusterId  = DiagnosticLogs::Id,
+            .fixedClusterServerEndpointCount =
+                1, // Diagnostic logs are device-wide functionality, so we only need one cluster instance
+            .maxEndpointCount = 1,
+        },
+        integrationDelegate);
+}
+
+void MatterDiagnosticLogsPluginServerInitCallback() {}
+void MatterDiagnosticLogsPluginServerShutdownCallback() {}
+
+namespace chip {
+namespace app {
+namespace Clusters {
+namespace DiagnosticLogs {
+
+DiagnosticLogsServer DiagnosticLogsServer::sInstance;
+
+DiagnosticLogsServer & DiagnosticLogsServer::Instance()
+{
+    return sInstance;
+}
+
+void DiagnosticLogsServer::SetDiagnosticLogsProviderDelegate(EndpointId endpoint, DiagnosticLogsProviderDelegate * delegate)
+{
+    VerifyOrReturn(endpoint == kRootEndpointId,
+                   ChipLogError(DeviceLayer, "DiagnosticLogsProviderDelegate can only be set on root endpoint"));
+    VerifyOrReturn(delegate != nullptr, ChipLogError(DeviceLayer, "DiagnosticLogsProviderDelegate cannot be nullptr"));
```

Comment:
why? I would drop the log and explain in a comment.

Does the delegate not start off as null? So why if I set it at start I cannot reset it back?